app-backend/app/src/test/scala/datasource/DatasourceSpec.scala

```diff
@@ -0,0 +1,156 @@
+    "contrast" -> 9,
+    "redBand" -> 3,
+    "greenGamma" -> 0
+  )
+
+  val datasource1 = Datasource.Create(
+    publicOrgId,
+    "Datasource1",
+    Visibility.Public,
+    exColorCorrect,
+    Map()
+  )
+
+  val datasource2 = Datasource.Create(
+    publicOrgId,
+    "Datasource2",
+    Visibility.Public,
+    exColorCorrect,
+    Map()
+  )
+
+  // A third datasource scoped to a different privacy level
+  val datasource3 = Datasource.Create(
+    publicOrgId,
+    "Datasource3",
+    Visibility.Private,
+    exColorCorrect,
+    Map()
+  )
+
+  // Alias to baseRoutes to be explicit
+  val baseRoutes = routes
+
+  "/api/datasources/" should {
+    "require authentication" in {
+      Get("/api/datasources/") ~> baseRoutes ~> check {
+        reject
+      }
+      Get("/api/datasources/").withHeaders(
+        List(authHeader)
+      ) ~> baseRoutes ~> check {
+        responseAs[PaginatedResponse[Datasource]]
+      }
+    }
+
+    "create an datasource successfully once authenticated" in {
+      Post("/api/datasources/").withHeadersAndEntity(
+        List(authHeader),
+        HttpEntity(
+          ContentTypes.`application/json`,
+          datasource1.toJson.toString()
+        )
+      ) ~> baseRoutes ~> check {
+        responseAs[Datasource]
+      }
+
+      Post("/api/datasources/").withHeadersAndEntity(
+        List(authHeader),
+        HttpEntity(
+          ContentTypes.`application/json`,
+          datasource2.toJson.toString()
+        )
+      ) ~> baseRoutes ~> check {
+        responseAs[Datasource]
+      }
+
+      Post("/api/datasources/").withHeadersAndEntity(
+        List(authHeader),
+        HttpEntity(
+          ContentTypes.`application/json`,
+          datasource3.toJson.toString()
+        )
+      ) ~> baseRoutes ~> check {
+        responseAs[Datasource]
+      }
+    }
+  }
+
+  /** Ignored because there's only one user + org in the db with datasources */
+  "scope to organization and privacy correctly" ignore {
+    Get(s"$baseDatasourcePath?organization=${publicOrgId}").withHeaders(
+      List(authHeader)
+    ) ~> baseRoutes ~> check {
+      // total is 4 because of landsat and sentinel in migration 38
+      responseAs[PaginatedResponse[Datasource]].count shouldEqual 3
+    }
+  }
+
+  "filter by name correctly" in {
+    val url1 = s"$baseDatasourcePath?name=${datasource1.name}"
+    Get(url1).withHeaders(
+      List(authHeader)
+    ) ~> baseRoutes ~> check {
+      responseAs[PaginatedResponse[Datasource]].count shouldEqual 1
+    }
+
+    val url2 = s"$baseDatasourcePath?name=${datasource2.name}"
+    Get(url2).withHeaders(
+      List(authHeader)
+    ) ~> baseRoutes ~> check {
+      responseAs[PaginatedResponse[Datasource]].count shouldEqual 1
+    }
+
+    val url3 = s"$baseDatasourcePath?name=${datasource3.name}"
+    Get(url3).withHeaders(
+      List(authHeader)
+    ) ~> baseRoutes ~> check {
+      responseAs[PaginatedResponse[Datasource]].count shouldEqual 0
```

Comment:
Yeah it should I was doing different things with privacy/org scopes before and forgot to change